cache.py

```diff
@@ -19,12 +20,13 @@ def add(self, url:str):
                         resolution="360p",
                         progressive=True,
                     ).order_by("resolution").desc().first()
-        video_file_name = video.default_filename
-        if (video_file_name not in os.listdir(self.file_path)):
-            video.download(self.file_path)
+        UUID = uuid.uuid4()
+        theuuid = str(UUID)
+        if (theuuid not in os.listdir(self.file_path)):
+            video.download(self.file_path,filename=theuuid+".mp4")
         video_id = self.get_video_id(url)
-        self.video_id_to_path[video_id] = os.path.join(self.file_path, video_file_name)
-        self.current_size_bytes += os.path.getsize(os.path.join(self.file_path, video_file_name))
+        self.video_id_to_path[video_id] = os.path.join(self.file_path, theuuid+".mp4")
+        self.current_size_bytes += os.path.getsize(os.path.join(self.file_path, theuuid +".mp4"))
         self._downsize_cache_to_target_bytes(self.max_size_bytes)
```

Comment:
i think what the code should do is 
1. download the file (already in the existing code)
2. rename the downloaded file path to the uuid string (we need to do this)
3. set `self.video_id_to_path[video_id]` to the file name

Reply:
done